_data-prepper/managing-data-prepper/latency.md

```diff
@@ -0,0 +1,145 @@
+---
+layout: default
+title: Pipeline latency tuning guide
+parent: Managing OpenSearch Data Prepper
+nav_order: 45
+---
+
+# Pipeline latency tuning guide
+
+This section collects the most impactful configuration for reducing end‑to‑end latency and provides copy‑paste examples. 
+
+“Latency” can mean:
+
+- **Ingest latency**: The time it takes from the source receiving data until the sink sends it to OpenSearch, or another destination.
+- **Searchable latency**: The time it takes until data becomes visible in OpenSearch search results. This is capped by the index’s `refresh_interval`.
+
+## Low Latency configuration
+
+The following table lists configuration that can be tweaked to improve latency.
+
+Component | Setting | Why it matters | Low‑latency starting point | Trade‑offs
+:--- | :--- | :--- | :--- | :---
+**Pipeline loop** | `workers` in each pipeline | More parallelism reduces queueing in CPU/IO bound pipelines. | The number of CPU cores, increase if sinks are I/O bound. | Higher CPU, more concurrent requests to sinks.
+**Pipeline loop** | `delay` | Sleep between buffer reads. | `0`–`10ms` to pull as soon as possible. | Lower delays increase polling overhead, increasing context switches and CPU wakeups. Tune to balance latency and CPU.
+**Bounded blocking buffer** | `batch_size` | Smaller batches flush sooner. | 64–256 | Smaller batches decrease throughput and increase request rate.
+**Peer Forwarder** | `batch_size`, `request_timeout` | The batch sizing and timeouts affect hop‑to‑hop delay. | Keep `batch_size` modest for example 48–128. | Too small `batch_size` reduces throughput. `request_timeout` too low increases retries/timeouts under load.
+**Peer Forwarder** | `forwarder` [configuration]({{site.url}}{{site.baseurl}}/data-prepper/managing-data-prepper/peer-forwarder/#configuration) | Caps queueing before forwarding. | Use low timeouts for example 50–200 ms. | Shorter timeouts can increase in-flight requests/connections, adding CPU, memory, TLS handshakes, and context-switch overhead. Too long causes queue build-up and higher tail latency.
+**Aggregate processors** | `group_duration` | The events wait for the window to close. | Prefer removing aggregation. If required, keep the window minimal, for example `5s`. | Smaller windows may break grouping semantics.
+**OpenSearch sink** | `bulk_size` (MiB) | Smaller bulks flush sooner. | 1–5 MiB | Very small `bulk_size` leads to more bulk requests for the same data, extra HTTP/TLS overhead, more threadpool contention, smaller Lucene batches, lower throughput. Very large leads to longer time to fill a batch, bigger retries, and memory spikes,and higher p95/p99.
+**OpenSearch side** | `index.refresh_interval` | Controls when data becomes searchable. | `1s` (default). | Lower refresh increases segment churn and indexing cost.
+
+The Delay processor adds latency by design. Avoid it in low‑latency pipelines.
```

Comment:
Is this the processor name? If so, should it be in code font (`delay` processor)?